### PR TITLE
Jonmv/reindexing data stores

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please reivew.

### CODE REVIEW

**Comment by jonmv** [comment]
PTAL.

File: configserver/src/main/java/com/yahoo/vespa/config/server/application/ConfigConvergenceChecker.java [hunk 1]
```diff
@@ -70,26 +70,30 @@ public ConfigConvergenceChecker(StateApiFactory stateApiFactory) {
...
                    .forEach(host -> host.getServices().stream()
                                         .filter(service -> serviceTypesToCheck.contains(service.getServiceType()))
                                         .forEach(service -> getStatePort(service).ifPresent(port -> servicesToCheck.add(service))));
 
-        Map<ServiceInfo, Long> currentGenerations = getServiceGenerations(servicesToCheck, timeoutPerService);
+        return getServiceGenerations(servicesToCheck, timeoutPerService);
+    }
+
+    /** Check all services in given application. Returns the minimum current generation of all services */
+    public ServiceListResponse getServiceConfigGenerationsJson(Application application, URI requestUrl, Duration timeoutPerService) {
```
Comment (bjorncs) [inline configserver/src/main/java/com/yahoo/vespa/config/server/application/ConfigConvergenceChecker.java:85]: Consider renaming to `getServiceConfigGenerationsResponse`

File: configserver/src/main/java/com/yahoo/vespa/config/server/application/ReindexingStatus.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+public class ReindexingStatus implements Reindexing {
+
+    private static final ReindexingStatus empty = new ReindexingStatus(Map.of(), Map.of());
+
+    private final Map<String, Long> pending;
+    private final Map<String, Status> ready;
+
+    ReindexingStatus(Map<String, Long> pending, Map<String, Status> ready) {
+        this.pending = Map.copyOf(pending);
+        this.ready = ready;
```
Comment (bjorncs) [inline configserver/src/main/java/com/yahoo/vespa/config/server/application/ReindexingStatus.java:28]: What is the reason for not copying the second parameter?

**Review by bjorncs — APPROVED** [review]
👍

File: configserver/src/main/java/com/yahoo/vespa/config/server/application/ReindexingStatus.java (hunk 2 above)
Comment (jonmv) [reply configserver/src/main/java/com/yahoo/vespa/config/server/application/ReindexingStatus.java:28]: Silly author, most likely >_<

**Comment by jonmv** [comment]
Thanks.

**Pull request merged**